src/surface.cpp

```diff
@@ -85,6 +87,25 @@ void read_coeffs(pugi::xml_node surf_node, int surf_id, double& c1, double& c2,
   }
 }
 
+void read_coeffs(pugi::xml_node surf_node, int surf_id, double& c1, double& c2,
+  double& c3, double& c4, double& c5, double& c6)
+{
+  // Check the given number of coefficients.
+  std::string coeffs = get_node_value(surf_node, "coeffs");
+  int n_words = word_count(coeffs);
+  if (n_words != 6) {
+    fatal_error(fmt::format(
+      "Surface {} expects 6 coeffs but was given {}", surf_id, n_words));
+  }
+
+  // Parse the coefficients.
+  int stat = sscanf(
+    coeffs.c_str(), "%lf %lf %lf %lf %lf %lf", &c1, &c2, &c3, &c4, &c5, &c6);
+  if (stat != 6) {
+    fatal_error("Something went wrong reading surface coeffs");
```

Comment:
Since we have the info here, we may as well report the surface ID in the error message too.